### PR TITLE
Use cache for subnets and compute instances

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
Follow up of https://github.com/elastic/assetbeat/pull/249

This PR implements caching for GCP Subnets and Compute Instances.

This way assets required by multiple asset type collectors can be retrieved from the cache instead of making extra API calls.
This PR also updates the compute instances parents to use subnets ids instead of vpc.
This is aligned with AWS collector.

Closes https://github.com/elastic/assetbeat/issues/254

### CODE REVIEW

File: input/gcp/gcp.go [hunk 1]
```diff
@@ -58,7 +58,9 @@ func configure(cfg *conf.C) (stateless.Input, error) {
 
 func newAssetsGCP(config config) (*assetsGCP, error) {
 	vpcAssetsCache, _ := freelru.New[string, *vpc](8192, hashStringXXHASH)
-	return &assetsGCP{config, vpcAssetsCache}, nil
+	subnetAssetsCache, _ := freelru.New[string, *subnet](8192, hashStringXXHASH)
+	computeAssetsCache, _ := freelru.New[string, *computeInstance](8192, hashStringXXHASH)
```
Comment (girodav) [inline input/gcp/gcp.go:62]: NIT: We could use the new `getEmpty....` functions that you defined in `util.go` here

Comment (MichaelKatsoulis) [reply input/gcp/gcp.go:62]: I thought about it but I didn't do it in the end. Will do

File: input/gcp/util.go [hunk 2]
```diff
@@ -89,3 +100,47 @@ func wantZone(zone string, confRegions []string) bool {
...
+					Key:   proto.String("kube-labels"),
+					Value: proto.String("cloud.google.com/gke-nodepool=mynodepool"),
+				},
+			},
+		},
+	}
+	selfLink := "https://www.googleapis.com/compute/v1/projects/elastic-observability/zones/europe-west1-d/instances/my-instance-1"
+	computeAssetsCache.AddWithExpire(selfLink, &cI, 60*time.Second)
+	return computeAssetsCache
+}
```
[the remark this answers is not in the document]
Comment (MichaelKatsoulis) [reply input/gcp/util.go:141]: I initially had them inside the tests but then they were called by multiple test files and I decided to move them in a more centra location. Maybe renaming them to getTestComputeCache is better.

**Review by girodav — APPROVED** [review]
LGTM ! 🚀

**Pull request merged**